crypto/s2n_libcrypto.c

```diff
@@ -134,6 +138,19 @@ bool s2n_libcrypto_is_boringssl()
 #endif
 }
 
+S2N_RESULT s2n_libcrypto_init(void)
+{
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
```

Comment:
Also test if the crypto library has been built with RC4 support LIBCRYPTO_SUPPORTS_EVP_RC4?
Otherwise RC4 is a hard requirement for OpenSSL 3.0.

Reply:
added check in https://github.com/aws/s2n-tls/pull/3457/commits/3a5f52ca0fc65e51e9cf31ec5fde11b5cbbb8c67